Assets/Scripts/GameController.cs

```diff
@@ -83,5 +106,111 @@ public void GameOver()
+        uiOkButton.onClick.AddListener(TaskOnClick);
+    }
+
+    void TaskOnClick() 
+    {
+        enterNameUI.SetActive(false);
+
+        if (PlayerPrefs.HasKey("highscoreTable"))
+        {
+            AddHighScore(score, enterNameText.text);
+            MakeHighScoreTable();
+        } 
+        else // highscoreTable hasn't been created yet 
+        {
+            MakeHighScoreTable();
+        }
+
+        highscoreTemplate.SetActive(true);
+        rollHighScore = true;
+    }
+
+    void MakeHighScoreTable() 
+    {
+        Highscores highscores;
+        string jsonString;
+        if (PlayerPrefs.HasKey("highscoreTable"))
+        {
+            jsonString = PlayerPrefs.GetString("highscoreTable");
+            highscores = JsonUtility.FromJson<Highscores>(jsonString);
+
+            for (int i = 0; i < highscores.highscoreList.Count; i++) 
+            {
+                for (int j = i + 1; j < highscores.highscoreList.Count; j++)
+                {
+                    if (highscores.highscoreList[j].score > highscores.highscoreList[i].score)
+                    {
+                        Highscore swap = highscores.highscoreList[i];
+                        highscores.highscoreList[i] = highscores.highscoreList[j];
+                        highscores.highscoreList[j] = swap;
+                    }
+                }
+            }
+        } 
+        else 
+        {
+            // Manually create list if it isn't stored in PlayerPrefs yet
+            highscoreList = new List<Highscore>() 
+            {
+                new Highscore { score = score, name = enterNameText.text }
+            };
+
+            highscores = new Highscores { highscoreList = highscoreList };
+            string json = JsonUtility.ToJson(highscores);
+            
+            PlayerPrefs.SetString("highscoreTable", json);
+            PlayerPrefs.Save();
+            
+            jsonString = PlayerPrefs.GetString("highscoreTable");
+            highscores = JsonUtility.FromJson<Highscores>(jsonString);
+        }
+
+        highscoreTransformList = new List<Transform>();
+        
+        foreach (Highscore highscore in highscores.highscoreList) 
+        {
+            AddHighScoreTransform(highscore, templateContainer, highscoreTransformList);
+        }    
+    }
+
+    private void AddHighScoreTransform(Highscore highscoreEntry, Transform scoreContainer, List<Transform> transformList)
+    {
+        Transform newRow = Instantiate(template, templateContainer);
+        newRow.localPosition = new Vector2 (0, -45 * transformList.Count);
+        newRow.Find("HighscorePlaceTemplate").GetComponent<Text>().text = (transformList.Count + 1).ToString();
+        newRow.Find("HighscoreNameTemplate").GetComponent<Text>().text = (highscoreEntry.name);
+        newRow.Find("HighscoreScoreTemplate").GetComponent<Text>().text = (highscoreEntry.score).ToString();
+        transformList.Add(newRow);

```

Comment:
I really like that you're dynamically creating the score board.
It would be a little cleaner (and a little more efficient) if maybe the newRow had its own class or struct or something so you could refer to HighscorePlace, HighscoreName, and HighscoreScore instances in that object rather than running a Find() on each of them for every row.
Generally Find() is pretty expensive, and there is usually a better way to do things. If you use it here and there that's fine, but always look for ways to optimize instead of using it--especially in methods that may run many times.